### PR TITLE
Homework3

### PR DESCRIPTION
**Pull request by majkel84**
1. TimeMock
2. Test to duplicate main
3. module3.yml

### CODE REVIEW

File: test/mocks/PizzaMock.hpp [hunk 1]
```diff
@@ -26,8 +26,11 @@ struct PizzaStub : public Pizza
 struct PizzaMock : public Pizza
 {
     PizzaMock() : Pizza("mock", 0.0, minutes(0)) {} // constructors should be avoided. Interface is advised
-    MOCK_CONST_METHOD0(getName, std::string());
-    MOCK_CONST_METHOD0(getPrice, double());
-    MOCK_CONST_METHOD0(getBakingTime, minutes());
+    //MOCK_CONST_METHOD0(getName, std::string());
+    //MOCK_CONST_METHOD0(getPrice, double());
+    //MOCK_CONST_METHOD0(getBakingTime, minutes());
```
Comment (Kasperice) [inline test/mocks/PizzaMock.hpp:31]: I think you can delete those comments

**Review by Kasperice — APPROVED** [review]
Nice code, the only thing you could do is to clean it from non required comments

File: test/mocks/PizzaMock.hpp (hunk 1 above)
Comment (majkel84) [reply test/mocks/PizzaMock.hpp:31]: I've left it for learning purpose

Comment (Kasperice) [reply test/mocks/PizzaMock.hpp:31]: ok :)

File: src/Pizzeria.cpp [hunk 2]
```diff
@@ -49,11 +49,14 @@ void Pizzeria::bakePizzas(int orderId)
         for (const auto & pizza : pizzas)
         {
             std::cout << "Baking " << pizza->getName() << std::endl;
-            std::this_thread::sleep_for(pizza->getBakingTime());
+            //std::this_thread::sleep_for(pizza->getBakingTime());
```
Comment (WBigaj) [inline src/Pizzeria.cpp:52]: Is this comment necessary?

File: test/PizzeriaTest.cpp [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #include <gtest/gtest.h>
 #include <string>
+//#include "mocks/DummyTimer.hpp"
```
Comment (WBigaj) [inline test/PizzeriaTest.cpp:3]: Another one.

File: test/PizzeriaTest.cpp [hunk 4]
```diff
@@ -11,7 +13,10 @@ using namespace ::testing;
 struct PizzeriaTest : public ::testing::Test
 {
 public:
-    Pizzeria pizzeria = Pizzeria("dummyName"); 
+    //DummyTimer dt;
+    //Pizzeria pizzeria = Pizzeria("dummyName", dt);
```
Comment (WBigaj) [inline test/PizzeriaTest.cpp:17]: Similar here.

**Review by WBigaj — COMMENTED** [review]
I have checked your code and do you really need those comment in the program?

File: src/Pizzeria.cpp (hunk 2 above)
Comment (majkel84) [reply src/Pizzeria.cpp:52]: I've left it for learning purpose, to checj how it looks in dummy style

File: test/PizzeriaTest.cpp (hunk 3 above)
Comment (majkel84) [reply test/PizzeriaTest.cpp:3]: Same in here

File: test/PizzeriaTest.cpp (hunk 4 above)
Comment (majkel84) [reply test/PizzeriaTest.cpp:17]: Same in here

File: src/Pizzeria.cpp (hunk 2 above)
Comment (WBigaj) [reply src/Pizzeria.cpp:52]: Isn't branching in Github designed for purpose of switching between versions of code? I think that's better to have two working version for learning purposes than having commented code. If you return there after long time you would need some time to figure what to comment/uncomment to get alternate version.

File: test/PizzeriaTest.cpp (hunk 4 above)
Comment (majkel84) [reply test/PizzeriaTest.cpp]: done

File: test/PizzeriaTest.cpp (hunk 3 above)
Comment (majkel84) [reply test/PizzeriaTest.cpp]: done

File: src/Pizzeria.cpp (hunk 2 above)
Comment (majkel84) [reply src/Pizzeria.cpp]: done

File: test/PizzeriaTest.cpp [hunk 5]
```diff
@@ -52,16 +56,46 @@ TEST_F(PizzeriaTest, completeOrderWithStubPizza)
...
+    EXPECT_CALL(mockTimer, sleep_for(minutes(9))).Times(1);
+
+    // When
+    auto orderId = pizzeria.makeOrder(pizzas);
+    auto price = pizzeria.calculatePrice(orderId);
+    pizzeria.bakePizzas(orderId);
+    pizzeria.completeOrder(orderId);
+
+    // Then
+    ASSERT_EQ(75, price);
```
Comment (ziobron) [inline test/PizzeriaTest.cpp:100]: You have some magic values here like 40.0, 25.0, 1, 5, 9. Use constexpr and name them properly. Then you should use them in proper EXPECT_CALL or ASSERT_EQ to show connections. Which value will work in which place.

**Review by ziobron — APPROVED** [review]

**Comment by ziobron** [comment]
{ "requirement_ids" : [1119, 1119, 1119, 3524, 3525, 3526] }

3 * 3 XP za punktualność (1119)
10 XP za TimeMock w Pizzas (3524)
15 XP za nowy przypadek testowy w Pizzas (3525)
5 XP za workflow w Pizzas (3526)

**Pull request closed without merging**